src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Unix.cs

```diff
@@ -5,13 +5,28 @@
 using System.ComponentModel;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.InteropServices;
 
 namespace System.IO.Ports
 {
     public partial class SerialPort : Component
     {
         public static string[] GetPortNames()
         {
+#if NETCOREAPP
+            return OperatingSystem.IsLinux() ? GetPortNames_Linux()
+                : OperatingSystem.IsMacOS() ? GetPortNames_OSX()
+                : OperatingSystem.IsFreeBSD() ? GetPortNames_FreeBSD()
+#else
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? GetPortNames_Linux()
```

Comment:
Won't FreeBSD pass the `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)` check?